x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuid from 'uuid';
+import { AlertEvent, EndpointEvent, EndpointMetadata, OSFields } from './types';
+
+export type Event = AlertEvent | EndpointEvent;
+
+interface EventOptions {
+  timestamp: number;
+  entityID?: string;
+  parentEntityID?: string;
+  eventType?: string;
```

Comment:
Does it make sense to have these types be derived from `EndpointEvent` types? If we did that, if the `event.type` changes in the future, we wouldn't have to update both `EndpointEvent` AND `EventOptions`.  It could be something like:
```
eventType?: EndpointEvent['event']['type']
```